ofctrl/ofctrl.go

```diff
@@ -99,15 +126,80 @@ func (c *Controller) Listen(port string) {
 
 }
 
+// Connect to Unix Domain Socket file
+func (c *Controller) Connect(sock string) {
+	if c.stopFlag == nil {
+		// Construct stop flag for notifying controller to stop connections
+		c.stopFlag = make(chan bool)
+		// Reset connection mode as ClientMode
+		c.connectMode = ClientMode
+	}
+	if c.disconFlag == nil {
+		// Construct disconnection flag for notifying controller to retry connections
+		c.disconFlag = make(chan bool)
+	}
+
+	go func() {
+		// Setup initial connection
+		c.disconFlag <- true
+	}()
+
+	var conn net.Conn
+	var err error
+	defer func() {
+		if conn != nil {
+			conn.Close()
+		}
+	}()
+
+	for {
+		select {
+		case <-c.stopFlag:
+			log.Println("Controller is delete")
+			return
+		case disConnection := <-c.disconFlag:
+			if disConnection == false {
+				continue
+			}
+			log.Printf("%s is disconnected, connecting...", sock)
+
+			if conn != nil {
+				// Try to close existent connection
+				_ = conn.Close()
+			}
+
+			conn, err = net.Dial("unix", sock)
+			if err != nil {
+				log.Fatal(err)
```

Comment:
Is it really a good idea to kill the controller if the connection fails?

Reply:
I would add retry here. However, if the connection still fails after max retry, I think we should kill the controller as it is not available. What do you think?